mage_ai/server/data/models.py

```diff
@@ -226,6 +227,12 @@ def __init__(self, id=None, path=None, feature_set_id=None, pipeline=None):
         if pipeline is not None:
             self.pipeline = pipeline
 
+        if api_key is not None:
```

Comment:
Is this change needed?

Reply:
If the api_key is None then I think it makes sense to not try to sync pipelines

Reply:
You can initialize the `Pipeline` model class with a `pipeline`, so I guess I was thinking in that case, we would want to sync it. However, I don't think we actually ever initialize a `Pipeline` like that, so we could remove that if it's not necessary.

Reply:
ok, that makes sense. maybe change it to `pipeline is not None and api_key is not None`